esx_service/utils/kvESX.py

```diff
@@ -42,24 +47,78 @@
 # Default kv side car alignment
 KV_ALIGN = 4096
 
+global using_64bits
+
 # Load the disk lib API library
-def loadDiskLib():
+def loadDiskLib(lib_name):
     global lib
 
     if not lib:
-        lib = CDLL(DISK_LIB)
-        lib.DiskLib_Init.argtypes = []
-        lib.DiskLib_Init.restype = c_bool
-        lib.DiskLib_Init()
+       lib = CDLL(lib_name)
+       lib.DiskLib_Init.argtypes = []
+       lib.DiskLib_Init.restype = c_bool
+       lib.DiskLib_Init()
 
 
 # Loads the ESX library used to access proprietary ESX disk lib API.
 # Create arg/result definitions for those that we use.
-def kvESXInit():
+
+# Initialize 32-bit disk lib interfaces
+def kv_ESX_lib64_init():
```

Comment:
Args are 64-bit handles passed to disklib in the 64-bit init and 32-bit
handles otherwise. I didn't want to have one large function with most
functions declared in a if - else statement. I can separate out the ones
that are not impacted into a kv_esx_lib_common() so that just the 32-64-bit
declarations are in one function (separated of course).

On Tue, Jun 28, 2016 at 3:14 AM, Mark Sterin <notifications@github.com>
wrote:

> In esx_service/utils/kvESX.py
> <https://github.com/vmware/docker-volume-vsphere/pull/479#discussion_r68662102>
> :
>
> >
> >
> >  # Loads the ESX library used to access proprietary ESX disk lib API.
> >  # Create arg/result definitions for those that we use.
> > -def kvESXInit():
> > +
> > +# Initialize 32-bit disk lib interfaces
> > +def kv_ESX_lib64_init():
>
> why 64 bit init is different from 32 bit init? Looking into function
> defintion, they seem to using 32 or 64 bit explicitly, so why different
> functions?
>
> [nit] if it *is* needed, I'd prefer to hide this in kv_esx_lib_init()
> rather that having it done upstairs
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vmware/docker-volume-vsphere/pull/479/files/618f13068273be9c4251cba120e77e704c061ed4#r68662102>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APHseImn0eDt8YCRVocgYyfzflLbBf0Lks5qQERXgaJpZM4I4qrt>
> .
>